view/editor/html.php

```diff
@@ -24,12 +24,12 @@ protected function _initialize(KObjectConfig $config)
             'name'    => null,
             'attribs' => [],
             'options' => [
-                'baseHref'          => $this->getObject('request')->getSiteUrl().'/',
-                'language'          => substr($locale, 0, strpos( $locale, '-' )),
-                'contentsLanguage'  => substr($locale, 0, strpos( $locale, '-' )),
-                'height'            => '',
-                'width'             => '',
-                'extraAllowedContent'  => 'hr[id]',
+                'baseHref'             => $this->getObject('request')->getSiteUrl().'/',
```

Comment:
@jebbdomingo Shouldn't try to use request in a view. It creates unneeded coupling. 

This is a view right, can we not use the view url, which is based on the siteurl or would that pose a problem when the editor is used in the admin? If so the baseurl for the editor might be best set as a conf option when the editor is included in an other view?

Reply:
@johanjanssens 
What do you mean `view url`? 
We just need the baseHref for CKEditor to render the images in the editor correctly 
`getObject('request')->getSiteUrl()` or `getObject('request')->getBaseUrl()` returns it more simple e.g. http://joomla.box/timble or http://timble.dev
than the `KViewAbstract::getUrl()` which returns http://joomla.box:8080/timble/component/textman/?view=article&id=32&menu_id=101&layout=form/ instead and need more processing to get the correct base path.

We could simply use `JUri::base()` for baseHref and for special purposes like rendering in the admin-side, it can be simply overriden by passing the baseHref as a configuration option.

Reply:
@jebbdomingo We don't want to use Joomla code, only if absolutely necessary. We also don't like to create coupling between request and view. 

Right now the editor is only used in the frontend, so using KViewAbstract::getUrl()  to get the url we need doesn't pose any problems? We just need to do KViewAbstract::getUrl() and then do a toString() specifying BASE constant. See: https://github.com/joomlatools/joomlatools-framework/blob/master/code/libraries/joomlatools/library/http/url/url.php#L91